docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,270 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high-quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in `CLIP-as-service` can support indexing a very large dataset (millions/billions) and querying within 50ms, depending on the machine.
+
+In order to implement retrieval, we add an extra indexer after the encoder in CLIP-as-service. We use [`AnnLite`](https://github.com/jina-ai/annlite) in this case. 
+
+
+## Fast search in CLIP-as-service
+Index and search are easy in `CLIP-as-service`:
+
+```python
+from clip_client import Client
+
+client = Client('grpc://0.0.0.0:23456')
+
+# index
+r = client.index(
+    [
+        'she smiled, with pain',  # text
+        'https://clip-as-service.jina.ai/_static/favicon.png',  # image
+    ]
```

Comment:
would it be better to use docarray?
and add one or two more candidates

Reply:
This is the example from cas client api document